### ISSUE TITLE
check_snmp offset + -w -c

### ISSUE THREAD

**Issue by danja95** [nagios-plugins/nagios-plugins#496]
Hello,

i had complied the fork of check_snmp with offset fix, later i discoverd critical and warning are not working correctly:
```
root@icinga:/etc/icinga2/zones.d# /usr/lib/nagios/plugins/check_snmp -H 192.168.1.1 -C public -o 1.3.6.1.2.1.10.94.1.1.4.1.2.4 --offset=10000 -w 9000:9999 -c 7500:9999
SNMP OK - 9975 | iso.3.6.1.2.1.10.94.1.1.4.1.2.4=9975;9999;9999;
root@icinga:/etc/icinga2/zones.d# /usr/lib/nagios/plugins/check_snmp -H 192.168.1.1 -C public -o 1.3.6.1.2.1.10.94.1.1.4.1.2.4 --offset=10000 -w 9000: -c 7500:
SNMP OK - 9975 | iso.3.6.1.2.1.10.94.1.1.4.1.2.4=9975;0;0;
root@icinga:/etc/icinga2/zones.d# /usr/lib/nagios/plugins/check_snmp -H 192.168.1.1 -C public -o 1.3.6.1.2.1.10.94.1.1.4.1.2.4 --offset=10000 -w 9976: -c 7500:
SNMP WARNING - *9975* | iso.3.6.1.2.1.10.94.1.1.4.1.2.4=9975;0;0;
root@icinga:/etc/icinga2/zones.d# /usr/lib/nagios/plugins/check_snmp -H 192.168.1.1 -C public -o 1.3.6.1.2.1.10.94.1.1.4.1.2.4 --offset= -w 9976: -c 7500:
SNMP OK - inf | iso.3.6.1.2.1.10.94.1.1.4.1.2.4=inf;0;0;
```

original check_snmp from debian buster repo:
```
root@icinga:/etc/icinga2/zones.d# /usr/lib/nagios/plugins/check_snmp -H 192.168.1.1 -C public -o 1.3.6.1.2.1.10.94.1.1.4.1.2.4 --offset=1000 -w 9976: -c 7500:
SNMP OK - 99752000 | iso.3.6.1.2.1.10.94.1.1.4.1.2.4=99752000;9976:;7500:
```

i want to do:
Lower that int (99752000) to 99.75 , to make it more human readable.

Pls fix, or tell me what i do wrong.

**Comment by sawolf** [comment]
I don't know what you mean when you say "critical and warning are not working correctly". All 5 of the commands that you pasted here conform to the [threshold specification](http://nagios-plugins.org/doc/guidelines.html#THRESHOLDFORMAT).

If you downloaded a fork of the nagios-plugins project, I can't really help you there. You'll need to contact the author of that fork.

As for the behavior of `--offset`, the defined behavior for the version of check_snmp available in this repo is that it adds or subtracts a number from the returned value. It does not perform multiplication/division/scaling. It seems like you're looking for a multiplier to scale your result, so if anything you would be looking for new functionality.

**Comment by danja95** [comment]
yeah, i need some sort of scaling. is it possible to do this, without create a whole ne script?

**Comment by sawolf** [comment]
There's not currently a way to do this with that plugin. However, if you wanted to change the plugin to be able to do that, I'd be happy to accept a PR that adds an option for this. I think the most relevant code would be around line 500 or so of check_snmp.c

**Comment by box293** [comment]
I'm thinking @hedenface had already developed this ready for the next major release of plugins, i.e. 3.x. I think it's in a private commit somewhere.

**Comment by sawolf** [comment]
I don't believe there was any hand-off of his local repos when he left, so I'll leave this open as a feature request for now.
